src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -211,152 +207,80 @@ public bool TryGetTelemetryId(out Guid telemetryId)
                 {
-                    return allActionSets;
+                    applicableToSpan = unifiedSuggestedActionSet.ApplicableToSpan.Value.ToSpan();
                 }
 
-                return allActionSets.SelectAsArray(InlineActions);
-            }
-
-            private SuggestedActionSet InlineActions(SuggestedActionSet actionSet)
-            {
-                using var newActionsDisposer = ArrayBuilder<ISuggestedAction>.GetInstance(out var newActions);
-                foreach (var action in actionSet.Actions)
-                {
-                    var actionWithNestedActions = action as SuggestedActionWithNestedActions;
+                return new SuggestedActionSet(
+                    categoryName: unifiedSuggestedActionSet.CategoryName,
+                    actions: suggestedActions,
+                    title: unifiedSuggestedActionSet.Title,
+                    priority: ConvertToSuggestedActionSetPriority(unifiedSuggestedActionSet.Priority),
+                    applicableToSpan: applicableToSpan);
 
-                    // Only inline if the underlying code action allows it.
-                    if (actionWithNestedActions?.CodeAction.IsInlinable == true)
+                // Local functions
+                ISuggestedAction ConvertToSuggestedAction(IUnifiedSuggestedAction unifiedSuggestedAction)
+                    => unifiedSuggestedAction switch
                     {
-                        newActions.AddRange(actionWithNestedActions.NestedActionSets.SelectMany(set => set.Actions));
-                    }
-                    else
+                        UnifiedCodeFixSuggestedAction codeFixAction => new CodeFixSuggestedAction(
+                            ThreadingContext, _owner, codeFixAction.Workspace, _subjectBuffer,
+                            codeFixAction.CodeFix, codeFixAction.Provider, codeFixAction.CodeAction,
+                            codeFixAction.FixAllFlavors == null ? null : ConvertToSuggestedActionSet(codeFixAction.FixAllFlavors)),
+                        UnifiedCodeRefactoringSuggestedAction codeRefactoringAction => new CodeRefactoringSuggestedAction(
+                            ThreadingContext, _owner, codeRefactoringAction.Workspace, _subjectBuffer,
+                            codeRefactoringAction.CodeRefactoringProvider, codeRefactoringAction.CodeAction),
+                        UnifiedFixAllSuggestedAction fixAllAction => new FixAllSuggestedAction(
+                            ThreadingContext, _owner, fixAllAction.Workspace, _subjectBuffer,
+                            fixAllAction.FixAllState, fixAllAction.Diagnostic, fixAllAction.CodeAction),
+                        UnifiedSuggestedActionWithNestedActions nestedAction => new SuggestedActionWithNestedActions(
+                            ThreadingContext, _owner, nestedAction.Workspace, _subjectBuffer,
+                            nestedAction.Provider ?? this, nestedAction.CodeAction,
```

Comment:
not certain if i know if the `nestedAction.Provider ?? this` part is correct.  is that trying to mirror what we did before?

Reply:
Yep. In the original SuggestedActionSource, the "provider" passed in to the SuggestedActionWithNestedActions constructor was either a CodeRefactoring.Provider, CodeFixCollection.Provider, or `this`. The first two can be passed into the UnifiedSuggestedActionWithNestedActions constructor, however `this` (i.e. SuggestedActionsSourceProvider) cannot be since it's in a different layer. So instead, `this` is represented by null in the new representation.